Debug log level

Atomfusion1 opened the pull request:
The goal was a way to add a quick way for players to control the amount of talk in the MQ window, 0 = mostly off but errors, 1 = standard (current) talk, 3,4 can be control more, or everything could be shifted in that 0 is no talk even errors 1 errors 2 normal 3 deeper and 4 deepest

RekkasGit commented:
This is probably going to take a discussion, as I think i want to handle this in a different way.

RekkasGit commented:
There is already a debug log system in place, the MQ.Writes are only used if you want to push something to the console on purpose. I would suggest  changing those to a different _log.Write and set their level accordingly

Atomfusion1 commented:
I dont care how you want it done, the goal is to be able to control the amount of spam per character MQ window from individual ini files. the current _log handles actual E3 issues , I would not want to tie E3 issues to normal screen output, for instance i do not want to see every time my war kicks or monk kicks .. but maybe i want to see the casting of one of my clerics or wizards when im setting up new spells, then when im done i can just stop that casting .. that way if actual errors do enter the console then you can see those

RekkasGit commented:
Yup, that is what i mean its already in place. Example: Casting.cs, where the ability check is done.

MQ.Write($"\ag{spell.CastName} \am{targetName} \ao{targetID}");

changes to 

_log.Write($"\ag{spell.CastName} \am{targetName} \ao{targetID}",Logging.LogLevels.Info);

then just change

Logging.MinLogLevelTolog = Logging.LogLevels.Info  as the default for everyone to keep things as they are
and then change it to be quite to 

Logging.MinLogLevelTolog = Logging.LogLevels.Error  as the default

via some command, example the /debug command changes the log levels and trace levels. It is in basics.cs

Atomfusion1 commented:
Reset Branch to Master, 
Implemented /Debug {trace,debug,info,error,off} ... 
when you type /Debug error you get error and above trace .. 
/debug trace .. you get trace debug info and error

added CharacterSettings 
"Debug Log Level(debug,info,error)" and set default / initial to info
can then be changed to error if you want less spam and in game you can use /debug info to turn it on per character 

did the bard.cs, casting.cs, assist.cs, loot.cs
changed from MQ.Write to _log.Write() and set most changes to info

Atomfusion1 commented on E3Next/E3Next.csproj:
Internal Changes 
Though removing the powershell.exe $(ProjectDir)CopyToEQFolder.ps1</PostBuildEvent>
will help new people build as it produces an error and you need to know where to delete it

RekkasGit commented:
will take a full look at this after I get the group loot stuff finished, hopefully this weekend.
